app/lib/account/backend.dart

```diff
@@ -177,7 +178,16 @@ class AccountBackend {
     if (auth == null) {
       return null;
     }
-    return await _lookupOrCreateUserByOauthUserId(auth);
+    final user = await _lookupOrCreateUserByOauthUserId(auth);
+    if (user.isDeleted) {
+      // We don't expect this to happen, but it might, e.g. when an account was
+      // created with a non-Google e-mail address, and later deleted, then
+      // re-created.
```

Comment:
I think this is the wrong place to do this check...

Inside: `_lookupOrCreateUserByOauthUserId` we do a lookup for `usersWithEmail`, this is probably the place where if the user we found an email for `isDeleted` then we should not associate the oauthId with said user.

This means that we can end up with the case where more than one user entity has the same email. But that should be fine.

Reply:
Good point, moved.